### DISCUSSION TITLE
Prompt version only create when different

### DISCUSSION THREAD

**Discussion by brightsparc** [langfuse/langfuse#2161] [Ideas]
### Describe the feature or potential improvement

I'm looking into the prompt management: https://langfuse.com/docs/prompts/get-started#createupdate-prompt and it seems that create_prompt will always create a new version regardless if the payload has changed.  

It would be nice if this behaviour more like git.  Have you considered maybe creating a hash of the contents, and only creating a new version when anything changes.  This would avoid the situation where a new version is created that is identical to the previous.

**Comment by marcklingen** [comment]
Thanks for raising this, @brightsparc. I see how this could be a better behavior for workflows where many new prompt versions are published via API, and some idempotency would be beneficial. Currently not on the roadmap yet, would love to track interest for this here!

**Comment by clemra** [comment]
Linking to [this blogpost](https://simon.podhajsky.net/blog/2024/langfuse-prompt-workflow/) for more on idempotency & a nice workaround.

**Reply by francosbenitez** [reply]
Amazing 🚀

**Comment by ksundarraj-c-rpx** [comment]
> ### Describe the feature or potential improvement
> I'm looking into the prompt management: https://langfuse.com/docs/prompts/get-started#createupdate-prompt and it seems that create_prompt will always create a new version regardless if the payload has changed.
> 
> It would be nice if this behaviour more like git. Have you considered maybe creating a hash of the contents, and only creating a new version when anything changes. This would avoid the situation where a new version is created that is identical to the previous.
> 
> ### Additional information
> _No response_

@marcklingen , is there any chance of making this feature happen near future. Because same prompt on all the versions during application runtime increases db storage.

**Reply by francosbenitez** [reply]
@ksundarraj-c-rpx Did you try handling it with a custom utils function that checks if the prompt already exists? I agree, though, that it would be great to have a built-in utility like `get_or_create_prompt` for this use case.

**Reply by ksundarraj-c-rpx** [reply]
Yeah just a string comparison but seems not working properly

**Reply by marcklingen** [reply]
Interesting, no immediate plans as this is a somewhat niche issue, open for contributions on this though if you want to contribute a helper function to the langfuse sdks
